File: kattis/LTAI-2022-04-05/B/basicprogramming1.cpp
#include <bits/stdc++.h>

using namespace std;

typedef int64_t i64;

int main() {
  i64 n, k, sum = 0, evenSum = 0;
  cin >> n >> k;
  vector<i64> a(n);
  for (i64 i = 0; i < n; i++) {
    cin >> a[i];
    sum += a[i];
    if (a[i] % 2 == 0) {
      evenSum += a[i];
    }
  }

  if (k == 1) {
    cout << 7 << endl;
    return 0;
  } else if (k == 2) {
    if (a[0] > a[1]) {
      cout << "Bigger" << endl;
    } else if (a[0] < a[1]) {
      cout << "Smaller" << endl;
    } else {
      cout << "Equal" << endl;
    }
    return 0;
  } else if (k == 3) {
    i64 ans;
    ans = (a[0] + a[1] + a[2]) - max(max(a[0], a[1]), a[2]) -
          min(min(a[0], a[1]), a[2]);
    cout << ans << endl;
    return 0;
  } else if (k == 4) {
    cout << sum << endl;
    return 0;
  } else if (k == 5) {
    cout << evenSum << endl;
    return 0;
  } else if (k == 6) {
    for (i64 i = 0; i < n; i++) {
      a[i] %= 26;
      cout << (char)(a[i] + 'a');
    }
    return 0;
  } else if (k == 7) {
    i64 i = 0, counter = 0;
    while (true) {
      i = a[i];
      if (i < 0 || i >= n) {
        cout << "Out" << endl;
        return 0;
      }
      if (counter > 1e7) {
        cout << "Cyclic" << endl;
        return 0;
      }

      if (i == n - 1) {
        cout << "Done" << endl;
        return 0;
      }

      counter++;
    }
  }

  return 0;
}

File: kattis/IX/LTAI-2022-05-09/A/basicprogramming1.py
def main():
    n, t = input().split()
    n = int(n)
    t = int(t)

    a = input().split()
    a = [int(i) for i in a]

    if t == 1:
        print("7\n")

    elif t == 2:
        if a[0] > a[1]:
            print("Bigger\n")
        elif a[0] < a[1]:
            print("Smaller\n")
        else:
            print("Equal\n")

    elif t == 3:
        mx = max(max(a[0], a[1]), a[2])
        mn = min(min(a[0], a[1]), a[2])
        print(a[0] + a[1] + a[2] - mx - mn, end="\n")

    elif t == 4:
        s = 0
        for x in a:
            s += x
        print(s, end="\n")

    elif t == 5:
        s = 0
        for x in a:
            if x % 2 == 0:
                s += x
        print(s, end="\n")

    elif t == 6:
        for x in a:
            x %= 26
            print(chr(x + 97), end="")
        print("\n")

    elif t == 7:
        i = 0
        count = 0
        while True:
            i = a[i]
            if i >= n:
                print("Out\n")
                break
            elif i == n - 1:
                print("Done\n")
                break
            elif count > 2 * n:
                print("Cyclic\n")
                break

            count += 1


if __name__ == "__main__":
    main()
